### PR TITLE
Update `web3-eth-contract` to use `web3-eth` methods

### LINKED ISSUE #4685: Update `web3-eth-contract` to use `web3-eth` methods
- [ ] [eth_call](https://github.com/ChainSafe/web3.js/pull/4678/files#diff-7fa8f782e68ce845dbb6e8446b86041697fa427db5485b9c63bf689d0b32ed81R215)
- [ ] [eth_sendTransaction](https://github.com/ChainSafe/web3.js/pull/4678/files#diff-7fa8f782e68ce845dbb6e8446b86041697fa427db5485b9c63bf689d0b32ed81R237)
- [ ] [Support EIP-1559 transactions parameters ](https://github.com/ChainSafe/web3.js/pull/4678#discussion_r782201138)for all contract calls
- [ ] Add extended validation to each contract public method parameters with `web3-validator`

Currently these methods are not implemented and are awaiting completion of #4623 and #4652

### PR DESCRIPTION
**Pull request by nazarhussain**
## Description

Please include a summary of the changes and be sure to follow our [Contribution Guidelines](../CONTRIBUTIONS.md).

Fixes #4685

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran `npm run dtslint` with success and extended the tests and types if necessary.
- [x] I ran `npm run test:unit` with success.
- [x] I ran `npm run test:cov` and my test cases cover all the lines and branches of the added code.
- [x] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [x] I have tested my code on the live network.
- [x] I have checked the Deploy Preview and it looks correct.
- [x] I have updated the `CHANGELOG.md` file in the root folder.

### CODE REVIEW

**Review by luu-alex — APPROVED** [review]

File: packages/web3-eth-contract/src/contract.ts [hunk 1]
```diff
@@ -297,10 +306,12 @@ export class Contract<Abi extends ContractAbi>
 					call: async (options?: PayableCallOptions, block?: BlockNumberOrTag) =>
 						this._contractMethodCall(abi, params, options, block),
 					// TODO: Use `web3-eth-tx` package to return `PromiEvent` instead.
```
Comment (Spacesai1or) [inline packages/web3-eth-contract/src/contract.ts]: Remove `TODO`?

File: packages/web3-eth-contract/src/types.ts [hunk 2]
```diff
@@ -43,14 +44,14 @@ export interface ContractInitOptions {
 export type TransactionReceipt = ReceiptInfo;
 
 export interface NonPayableCallOptions {
-	nonce?: Numbers;
-	chainId?: Numbers;
+	nonce?: string;
+	chainId?: string;
```
Comment (Spacesai1or) [inline packages/web3-eth-contract/src/types.ts]:
```suggestion
	nonce?: HexString;
	chainId?: HexString;
```

File: packages/web3-eth-contract/src/types.ts [hunk 3]
```diff
@@ -43,14 +44,14 @@ export interface ContractInitOptions {
...
 	from?: Address;
 	to?: Address;
 	data?: HexString;
 	gas?: string;
-	maxPriorityFeePerGas?: Numbers;
-	maxFeePerGas?: Numbers;
+	maxPriorityFeePerGas?: string;
+	maxFeePerGas?: string;
```
Comment (Spacesai1or) [inline packages/web3-eth-contract/src/types.ts]:
```suggestion
	maxPriorityFeePerGas?: HexString;
	maxFeePerGas?: HexString;
```

**Review by Spacesai1or — APPROVED** [review]
Small changes requested

**Pull request merged**